packages/arb-provider-go/test/connection_test.go

```diff
@@ -0,0 +1,280 @@
+	return hexutil.Encode(buf), nil
+}
+
+func RunValidators(t *testing.T) (*FibonacciSession, error) {
+	coordinatorKey := "ffb2b26161e081f0cdf9db67200ee0ce25499d5ee683180a9781e6cceb791c39"
+	followerKey := "979f020f6f6f71577c09db93ba944c89945f10fade64cfc7eb26137d5816fb76"
+	err := setupValidators(coordinatorKey, followerKey, t)
+	if err != nil {
+		t.Errorf("Validator setup error %v", err)
+		t.FailNow()
+		return nil, err
+	}
+	//setupValidators()
+
+	privateKeyBytes, _ := hex.DecodeString(coordinatorKey)
+	pubKey, err := _computePubKeyString(privateKeyBytes)
+	if err != nil {
+		t.Errorf("_computePubKeyString error %v", err)
+		return nil, err
+	}
+	conn, dialerr := goarbitrum.Dial("", privateKeyBytes, pubKey)
+	if dialerr != nil {
+		t.Errorf("Dial error %v", dialerr)
+		return nil, err
+	}
+	key1, err := crypto.HexToECDSA(coordinatorKey)
+	if err != nil {
+		t.Errorf("HexToECDSA error %v", err)
+		return nil, err
+	}
+	auth := bind.NewKeyedTransactor(key1)
+	auth.GasLimit = 100000000
+
+	//conn, err = goarbitrum.Dial("", privateKeyBytes, _computePubKeyString(privateKeyBytes))
+	var fibAddr common.Address
+	fibAddr = common.HexToAddress("0x895521964D724c8362A36608AAf09A3D7d0A0445")
+	fib, err := NewFibonacci(fibAddr, conn)
+	if err != nil {
+		t.Errorf("NewFibonacci error %v", err)
+		return nil, err
+	}
+
+	//Wrap the Token contract instance into a session
+	fibonacciSession := &FibonacciSession{
+		Contract: fib,
+		CallOpts: bind.CallOpts{
+			From: auth.From,
+		},
+		TransactOpts: *auth,
+	}
+
+	return fibonacciSession, nil
+}
+
+func TestFib(t *testing.T) {
+	session, err := RunValidators(t)
+	if err != nil {
+		t.Errorf("Validator setup error %v", err)
+	}
+
+	t.Run("TestFibResult", func(t *testing.T) {
+		fibsize := 15
+		fibnum := 11
+		fmt.Println("generating fib")
+		_, err := session.GenerateFib(big.NewInt(int64(fibsize)))
+		if err != nil {
+			t.Errorf("GenerateFib error %v", err)
+			return
+		}
+		fmt.Println("getting fib")
+		fibval, err := session.GetFib(big.NewInt(int64(fibnum)))
+		if err != nil {
+			t.Errorf("GetFib error %v", err)
+			return
+		}
+		if fibval.Cmp(big.NewInt(144)) != 0 { // 11th fibanocci number
+			t.Errorf("GetFib error - expected %v got %v", big.NewInt(int64(144)), fibval)
+		}
+		log.Printf("Fibonacci value number %v = %v", fibnum, fibval)
+
+	})
+	t.Run("TestEvent", func(t *testing.T) {
+		eventChan := StartEventListeners(session.Contract)
+
+		fibsize := 15
+		go func() {
+			time.Sleep(5 * time.Second)
+			fmt.Println("generating fib")
+			_, err := session.GenerateFib(big.NewInt(int64(fibsize)))
+			if err != nil {
+				t.Errorf("GenerateFib error %v", err)
+				return
+			}
+		}()
+
+		fmt.Println("waiting for event")
+	Loop:
+		for ev := range eventChan {
+			switch event := ev.(type) {
+			case *FibonacciTestEvent:
+				fmt.Printf("Received fibonacci test event %v", event.Number)
+				break Loop
+			case ListenerError:
+				t.Errorf("errorEvent %v %v", event.ListenerName, event.Err)
```

Comment:
This error and the default error should break out of the loop along with causing the error or else there's a chance of infinite loop.